crates/astria-proto/src/native/sequencer/v1alpha1.rs

```diff
@@ -1251,6 +1266,1158 @@ impl Display for IncorrectAddressLength {
+                .map_err(SequencerBlockError::SignedTransactionProtobufDecode)?;
+            let signed_tx = SignedTransaction::try_from_raw(raw_tx)
+                .map_err(SequencerBlockError::RawSignedTransactionConversion)?;
+            for action in signed_tx.transaction.actions {
+                if let Action::Sequence(SequenceAction {
+                    rollup_id,
+                    data,
+                }) = action
+                {
+                    let elem = rollup_transactions.entry(rollup_id).or_insert(vec![]);
+                    elem.push(data);
+                }
+            }
+        }
+        rollup_transactions.sort_unstable_keys();
+
+        if rollup_transactions_root
+            != derive_merkle_tree_from_rollup_txs(&rollup_transactions).root()
+        {
+            return Err(SequencerBlockError::RollupTransactionsRootDoesNotMatchReconstructed);
+        }
+
+        // ensure the rollup IDs commitment matches the one calculated from the rollup data
+        if rollup_ids_root != merkle::Tree::from_leaves(rollup_transactions.keys()).root() {
+            return Err(SequencerBlockError::RollupIdsRootDoesNotMatchReconstructed);
+        }
+
+        // action tree root is always the first tx in a block
+        let rollup_transactions_proof = tree.construct_proof(0).expect(
+            "the tree has at least one leaf; if this line is reached and `construct_proof` \
+             returns None it means that the short circuiting checks above it have been removed",
+        );
+
+        let rollup_ids_proof = tree.construct_proof(1).expect(
+            "the tree has at least two leaves; if this line is reached and `construct_proof` \
+             returns None it means that the short circuiting checks above it have been removed",
+        );
+
+        Ok(Self {
+            block_hash,
+            header,
+            rollup_transactions,
+            rollup_transactions_proof,
+            rollup_ids_proof,
+        })
+    }
+
+    /// Converts from the raw decoded protobuf representatin of this type.
+    ///
+    /// # Errors
+    /// TODO(https://github.com/astriaorg/astria/issues/612)
+    pub fn try_from_raw(raw: raw::SequencerBlock) -> Result<Self, SequencerBlockError> {
+        fn rollup_txs_to_tuple(
+            raw: raw::RollupTransactions,
+        ) -> Result<(RollupId, Vec<Vec<u8>>), IncorrectRollupIdLength> {
+            let id = RollupId::try_from_slice(&raw.id)?;
+            Ok((id, raw.transactions))
+        }
+
+        let raw::SequencerBlock {
+            header,
+            rollup_transactions,
+            rollup_transactions_proof,
+            rollup_ids_proof,
+        } = raw;
+
+        let rollup_transactions_proof = 'proof: {
+            let Some(rollup_transactions_proof) = rollup_transactions_proof else {
+                break 'proof Err(SequencerBlockError::FieldNotSet(
+                    "rollup_transactions_proof",
+                ));
```

Comment:
why the `'proof` syntax over just returning the error?

Reply:
There are two steps in this block: one ensures the protobuf value was set, the next one actually does the conversion.

These two steps belong together logically and I didn't want to declare variables at the function level namespace.

I also think this is neat.

Reply:
```
break 'proof Err(SequencerBlockError::FieldNotSet(
   "rollup_transactions_proof",
));
```

this is logically the same as:

```
return Err(SequencerBlockError::FieldNotSet(
   "rollup_transactions_proof",
));
```

right?  just checking for my own sanity lol